crates/ide/src/folding_ranges.rs

```diff
@@ -305,10 +305,10 @@ use std::io as iop;</fold>
 
 <fold imports>use std::mem;
 use std::f64;</fold>
-
-use std::collections::HashMap;
-// Some random comment
-use std::collections::VecDeque;
+            
+<fold imports>use std::collections::HashMap;
+<fold imports>// Some random comment
+use std::collections::VecDeque;</fold></fold>
 
```

Comment:
I unfortunately don't understand what is happening here now. For some reason the folds are nested here, but if these use statements were `mod` declarations then there only exists one fold. This makes me confused as I would believe that `mod` items and `use` items should be equal in this regard now. I guess I am missing something but I'm unsure what that is

Reply:
We have special-case for `use` items to fold *all* of them, like you do in intelliJ:

![image](https://user-images.githubusercontent.com/1711539/101388770-b36be100-38c0-11eb-814a-7553f1ac3106.png)

![image](https://user-images.githubusercontent.com/1711539/101388800-bc5cb280-38c0-11eb-9626-0f2242f255be.png)


I don't think you need to touch folds here in this PR?

Reply:
These changes caused the `test_fold_import_groups` test to fail because its folds became this:

```rust
<fold imports>use std::str;
use std::vec;
use std::io as iop;</fold>

<fold imports>use std::mem;
use std::f64;</fold>

<fold imports>use std::collections::HashMap;
<fold imports>// Some random comment
use std::collections::VecDeque;</fold></fold>

fn main() <fold block>{
}</fold>
```
from the prior expected output
```rust
<fold imports>use std::str;
use std::vec;
use std::io as iop;</fold>

<fold imports>use std::mem;
use std::f64;</fold>

use std::collections::HashMap;
// Some random comment
use std::collections::VecDeque;

fn main() <fold block>{
}</fold>"
```

The change is for the collections imports with the comment inbetween.
They seem very much wrong to me as they are very weirdly nested and should only emit one fold there if any.

Reply:
There is most likely a good way to solve this special case but I couldn't think of any at the time I was working on this. I'll take another look again at it in a bit to see if I can prevent this weirdness from happening here.

Reply:
Ah, right, I see now. Yeah, seems like the old code was buggy!

bors r+